bytebuffer-collections/src/main/java/io/druid/collections/spatial/search/PolygonBound.java

```diff
@@ -150,14 +149,11 @@ else if (intersection_x < coords[0]) {
     return oddNodes;
   }
 
-
-  private boolean isPointLayingOnHorizontalBound(int i, int j, float coords[]){
-    return ordinate[i] == ordinate[j] && ordinate[j] == coords[1] &&
-            ((abscissa[i] < abscissa[j] && abscissa[j] > coords[0] && abscissa[i] < coords[0])
-            || (abscissa[i] > abscissa[j] && abscissa[j] < coords[0] && abscissa[i] > coords[0]));
+  private boolean between(int i, int j, float coords[]) {
```

Comment:
I mean a different thing. You just renamed `isPointLayingOnHorizontalBound()` to `between()`, that actually makes it less clear, because the former name was more precise. I meant to extract a method
```java
private static boolean between(float a, float b, float x)
{
  if (a <= b) {
    return a <= x && x <= b;
  } else {
    // b < a
    return b <= x && x <= a;
  }
}
```

And then `isPointLayingOnHorizontalBound()` is defined as
```java
private boolean isPointLayingOnHorizontalBound(int i, int j, float coords[])
{
  return ordinate[i] == ordinate[j] && ordinate[j] == coords[1] &&
      between(abscissa[i], abscissa[j], coords[0]);
}
```